### PR TITLE
Add tenant and display_name config options

### PR DESCRIPTION
**Pull request by boltgolt**

### CODE REVIEW

File: toon.py [hunk 1]
```diff
@@ -81,12 +91,18 @@ class ToonDataStore:
...
-        toon = Toon(username, password, consumer_key, consumer_secret)
+        toon = Toon(username,
+                    password,
+                    consumer_key,
+                    consumer_secret,
+                    tenant_id=tenant,
+                    display_common_name=name)
 
         self.toon = toon
         self.gas = gas
```
Comment (boltgolt) [inline toon.py:111]: What do these undocumented config values actually do?

**Comment by Housebeer** [comment]
Hi @Boltgolt ,

I get the following error when I run your commit. It has a problem when parsing the config file

```
 toon: 
    - Invalid config for [toon]: string value is None for dictionary value @ data['toon']['display_name'].
       Got None. (See /home/homeassistant/.homeassistant/configuration.yaml, line 70). 
       Please check the docs at https://home-assistant.io/components/toon/
```


If I change 
```
CONF_NAME = 'display_name'
DEFAULT_NAME = None
```
to 
```
CONF_NAME = 'display_name'
DEFAULT_NAME = ''
```
the error disappear. Is this an error or am I doing something wrong?

File: toon.py (hunk 1 above)
Comment (costastf) [reply toon.py:111]: Sorry, i missed this! I have been quite busy and not paying the proper attention apparently. The tenant id is a value about the provider. For the Netherlands it is eneco but toon is a device by quby that is rebranded in other countries too so by being able to pass the tenant id explicitly this component works for the same device in spain too. As for the display_common_name it is the name of the device that can be retrieved by the app at "Your Toon" as display code and can be used by people with multiple devices(houses) on the same eneco contract.

**Comment by costastf** [comment]
@Housebeer this should work fine with None but maybe @Boltgolt could have a look at it. I still have not migrated to this newer component and still using the older one with the unofficial api.

File: toon.py (hunk 1 above)
Comment (costastf) [reply toon.py:111]: @Boltgolt do you want to have a look at the issue @Housebeer reports and if it is nothing I can merge?

Comment (boltgolt) [reply toon.py:111]: I was specifically asking about the "gas" and "solar" config options. As far as i can see they don't actually do anything?

**Comment by boltgolt** [comment]
@Housebeer thanks for reporting! Turns out that newer hass versions have a more strict policy on conifg types, my last push should fix this.

**Pull request merged**